python/lsst/ts/mtaos/mtaos.py

```diff
@@ -762,6 +844,103 @@ async def do_resetOffsetDOF(self, data: salobj.type_hints.BaseDdsDataType) -> No
             await self.pubEvent_degreeOfFreedom()
             await self.pubEvent_mirrorStresses()
 
+    def do_stopClosedLoop(self):
+        """Stop the closed loop operation."""
+        self.assert_enabled()
+        self._logExecFunc()
+
+        if self.close_loop_task.done():
+            self.log.warning("Closed loop already stopped.")
+            return
+
+        self.closed_loop_task.cancel()
+        await self.evt_closedLoopState.set_write(state=ClosedLoopState.IDLE)
+        try:
+            await self.closed_loop_task
+        except asyncio.CancelledError:
+            pass
+
+    def do_startClosedLoop(self):
+        """Start the closed loop operation."""
+        self.assert_enabled()
+        if self.closed_loop_task.done():
+            self.closed_loop_task = asyncio.create_task(self.run_closed_loop())
+        else:
+            self.log.info("Closed loop already running, nothing to do.")
+
+    async def run_closed_loop(self):
+        """Closed loop operation."""
+        self._logExecFunc()
+
+        self.oods.evt_imageInOODS.flush()
```

Comment:
So, instead of creating `self.oods` as a class attribute that will end up only being used here, consider creating it here instead. 

Basically do something like:

```
async def run_closed_loop(self):
    async with salobj.Remote("MTOODS" if self.camera_name == "LSSTCam" else "CCOODS") as oods:
        while self.summary_state == salobj.State.Enabled:
            ....
```